### DISCUSSION TITLE
Center wezterm when opening additional new window(s) (CMD+N)

### DISCUSSION THREAD

**Discussion by Hacksore** [wezterm/wezterm#5578] [Q&A] [answered]
### What Operating System(s) are you running on?

macOS

### WezTerm version

wezterm 20240203-110809-5046fc22

### Ask your question!

I have this scripture and seems to work fine for wezterm first spawning but if I CMD+N to create a new window it goes to a seemingly random position.

```lua
-- center
wezterm.on("gui-startup", function()
  local screen = wezterm.gui.screens().active
  local _, _, window = wezterm.mux.spawn_window({})

  local gui_window = window:gui_window()

  -- Set dimensions to 85% of current screen size
  -- The actual dimensions will be a bit bigger if we take into
  -- account the decorations on top
  local width = screen.width * 0.85
  local height = screen.height * 0.85

  gui_window:set_inner_size(width, height)

  -- Position the window at the center of the screen
  local dimensions = gui_window:get_dimensions()
  local x = (screen.width - dimensions.pixel_width) * 0.5
  local y = (screen.height - dimensions.pixel_height) * 0.5

  gui_window:set_position(x, y)
end)
```

demo of issue:

https://github.com/wez/wezterm/assets/996134/729a67f2-d453-42b5-88bf-a793ef31c2c5

**Comment by Hacksore** [comment]
paging the termfather himself!

@wez any ideas or pointers here?

**Reply by wez** [reply]
What you want is some kind of "gui-window-created" event, which we don't have right now.
You could approximate it for yourself by using the "update-status" event to resolve the mux-window-id of the window, and if you haven't seen it before (keep a list of things you've seen in [wezterm.GLOBAL](https://wezfurlong.org/wezterm/config/lua/wezterm/GLOBAL.html), then center the gui window.

**Reply by Hacksore** [reply]
will look into this and figure out if I can find a working scripture, thanks @wez ❤️.

Should i open a feature request for a `gui-window-created` event?

**Reply by wez** [reply]
yeah, I though we already had an issue tracking that event, but we don't.

You might want to try this in the meantime: https://github.com/wez/wezterm/issues/256#issuecomment-1501101484

**Reply by Hacksore** [reply, marked as answer]
Was able to find a working solution for this with what @wez suggestion above.

```lua
local function center_window_once(window)
  wezterm.GLOBAL.windows_centered = wezterm.GLOBAL.windows_centered or {}

  local window_id = window:window_id() .. ""
  if wezterm.GLOBAL.windows_centered[window_id] then
    return
  end

  local screen = wezterm.gui.screens().active

  local width = screen.width * 0.85
  local height = screen.height * 0.85

  window:set_inner_size(width, height)

  local dimensions = window:get_dimensions()
  local x = (screen.width - dimensions.pixel_width) * 0.5
  local y = (screen.height - dimensions.pixel_height) * 0.5

  wezterm.GLOBAL.windows_centered[window_id] = true

  window:set_position(x, y)
end

wezterm.on("update-status", function(window)
  center_window_once(window)
end)

```

**Comment by basokant** [comment]
commenting for reach 🔼
